org-code-javabuilder/media/src/main/java/org/code/media/Color.java

```diff
@@ -9,12 +11,18 @@
   /**
    * Creates a color from a string representation.
    *
-   * @param color the string name of the color
-   * @throws IllegalArgumentException if the value specifies an unsupported color name or illegal
-   *     hexadecimal value
+   * @param color the string name of the color (case insensitive).
+   * @throws IllegalArgumentException if the value specifies an unsupported color name
    */
   public Color(String color) throws IllegalArgumentException {
-    // TODO: Implement this method
+    String colorAllCaps = color.toUpperCase();
+    if (!ColorConstantMap.containsKey(colorAllCaps)) {
+      throw new IllegalArgumentException("Invalid color " + color);
+    }
+    Color colorToCopy = ColorConstantMap.get(colorAllCaps);
+    this.red = colorToCopy.getRed();
+    this.green = colorToCopy.getGreen();
+    this.blue = colorToCopy.getBlue();
```

Comment:
nit: this can just be `this(colorToCopy)` right? or even `this(ColorConstantMap.get(colorAllCaps))`

Reply:
since we want to throw an exception if the color is invalid it doesn't work, because `this(...)` has to be the first line of the constructor.

Reply:
Ahh right yes, forgot about that :)